pkg/plugin/windows/hnsstats/hnsstats_windows.go

```diff
@@ -96,34 +97,39 @@ func pullHnsStats(ctx context.Context, h *hnsstats) error {
 			if err != nil {
 				h.l.Error("Getting Vswitch ports failed", zap.Error(err))
 			}
+
 			for _, ep := range endpoints {
 				if len(ep.IpConfigurations) < 1 {
 					h.l.Info("Skipping endpoint without IPAddress", zap.String("EndpointID", ep.Id))
 					continue
 				}
-				var mac string = ep.MacAddress
-				var ip string = ep.IpConfigurations[0].IpAddress
-				// var id string = ep.Id
-				if stats, err := hcsshim.GetHNSEndpointStats(ep.Id); err != nil {
-					h.l.Error("Getting endpoint stats failed for endpoint ID "+ep.Id, zap.Error(err))
+
+				id := ep.Id
+				mac := ep.MacAddress
+				ip := ep.IpConfigurations[0].IpAddress
+
+				if stats, err := hcsshim.GetHNSEndpointStats(id); err != nil {
+					h.l.Error("Getting endpoint stats failed for endpoint ID "+id, zap.Error(err))
```

Comment:
this is a zap logger, should not do string concat or interpolation but instead use k,v pairs:
```suggestion
					h.l.Error("Getting endpoint stats failed", zap.Error(err), zap.String("endpointID", id)
```

conceptually, this separates the _message_ of the log "getting endpoint stats failed", with the _data_ associated with that log "which endpoint, what error, etc"

Reply:
Thanks for spotting it, I'm updating all occurrences!